_rules/element-marked-decorative-is-not-exposed-46ca7f.md

```diff
@@ -0,0 +1,144 @@
+---
+id: 46ca7f
+name: Element marked as decorative is not exposed
+rule_type: atomic
+description: |
+  This rule checks that elements marked as decorative either are not included in the accessibility tree, or have a presentational role.
+accessibility_requirements:
+input_aspects:
+  - DOM tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+---
+
+## Applicability
+
+The rule applies to any element which is [marked as decorative][].
+
+## Expectation
+
+Each target element either is not [included in the accessibility tree][] or has a [semantic role][] of `none` or `presentation`.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+Implementation of the [Presentational Roles Conflict Resolution][] differs slightly from one user agent to the other. Hence, some elements might be exposed by one user agent and not by another, and consequently might create accessibility issues only for some users. In any cases, triggering the conflict is a sign of bad practice.
+
+## Background
+
+Elements are normally [marked as decorative][] to convey intention of the author that they are [pure decoration][decorative] and thus shouldn't be exposed by assistive technologies. On the other hand, elements that are [focusable][] are important to know for anybody and should be exposed to assistive technologies; and elements that are defining any [global ARIA attribute][] indicate intention to communicate something to the assistive technologies (through the `aria-*` attribute). When an element is both [marked as decorative][] and either [focusable][] or defining a [global ARIA attribute][], a conflict arise between these two intentions. The [conflict is resolved][presentational roles conflict resolution] by exposing the element.
+
+Whenever such a conflict occurs, this indicates at the very least mismatching intentions. Such a conflict should be avoided.
+
+When these conflicts arise on [decorative][] [non-text content][], this is also a failure of [Success Criterion 1.1.1: Non-text Content][sc111] because [decorative][] [non-text content][] must be implemented in a way that allows assistive technologies to ignore it. When this conflicts arise on text content, or on content which is not [decorative][], this is not a failure of WCAG. Therefore this rule is not mapping to any specific WCAG Success Criterion, and is not an accessibility requirement for WCAG.
+
+## Test Cases
```

Comment:
This is not something that will prevent me from accepting, but currently, the examples almost hint that this rule is applicable to images that are markd as decorative (all examples use `img` or `svg` elements). Given that the applicability is all elements that marked as decorative, we could have at least an example with text content.

Reply:
Good point. I'll add some.
The rule was initially only for images…